File: src/main/java/org/roborace/lapscounter/config/WebSocketConfig.java
package org.roborace.lapscounter.config;

import org.roborace.lapscounter.service.RoboraceWebSocketHandler;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.socket.config.annotation.EnableWebSocket;
import org.springframework.web.socket.config.annotation.WebSocketConfigurer;
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry;

@Configuration
@EnableWebSocket
public class WebSocketConfig implements WebSocketConfigurer {

    @Autowired
    private RoboraceWebSocketHandler roboraceWebSocketHandler;

    public void registerWebSocketHandlers(WebSocketHandlerRegistry registry) {
        registry.addHandler(this.roboraceWebSocketHandler, "/").setAllowedOrigins("*");
    }
}

File: src/main/kotlin/org/roborace/lapscounter/config/WebSocketConfig.kt
package org.roborace.lapscounter.config

import org.roborace.lapscounter.service.RoboraceWebSocketHandler
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Configuration
import org.springframework.web.socket.config.annotation.EnableWebSocket
import org.springframework.web.socket.config.annotation.WebSocketConfigurer
import org.springframework.web.socket.config.annotation.WebSocketHandlerRegistry

@Configuration
@EnableWebSocket
class WebSocketConfig : WebSocketConfigurer {
    @Autowired
    private lateinit var roboraceWebSocketHandler: RoboraceWebSocketHandler

    override fun registerWebSocketHandlers(registry: WebSocketHandlerRegistry) {
        registry.addHandler(roboraceWebSocketHandler, "/ws").setAllowedOrigins("*")
    }
}
